test/test_large_cohort.py

```diff
@@ -127,111 +141,297 @@ def _mock_cohort(dataset_id: str):
+        tmp_prefix=str(tmp_path / 'tmp'),
+        genome_build=conf['references']['genome_build'],
+        gvcf_paths=gvcf_paths,
+        vds_paths=None,
+        specific_intervals=conf['large_cohort']['combiner']['intervals'],
+    )
 
-        dense_mt_path = res_pref / 'dense.mt'
-        dense_subset.run(
-            vds_path=str(vds_path),
-            out_dense_mt_path=str(dense_mt_path),
-        )
+    # do some testing here
+    assert exists_not_cached(vds_path)
 
-        relateds_to_drop_ht_path = res_pref / 'relateds_to_drop.ht'
-        relatedness.run(
-            dense_mt_path=dense_mt_path,
-            sample_qc_ht_path=sample_qc_ht_path,
-            out_relatedness_ht_path=res_pref / 'relatedness.ht',
-            out_relateds_to_drop_ht_path=relateds_to_drop_ht_path,
-            tmp_prefix=res_pref / 'tmp',
-        )
 
-        scores_ht_path = res_pref / 'scores.ht'
-        eigenvalues_ht_path = res_pref / 'eigenvalues.ht'
-        loadings_ht_path = res_pref / 'loadings.ht'
-        inferred_pop_ht_path = res_pref / 'inferred_pop.ht'
-        ancestry_sample_qc_ht_path = res_pref / 'ancestry_sample_qc.ht'
-
-        ancestry_pca.run(
-            dense_mt_path=dense_mt_path,
-            sample_qc_ht_path=sample_qc_ht_path,
-            relateds_to_drop_ht_path=relateds_to_drop_ht_path,
-            tmp_prefix=res_pref / 'tmp',
-            out_scores_ht_path=scores_ht_path,
-            out_eigenvalues_ht_path=eigenvalues_ht_path,
-            out_loadings_ht_path=loadings_ht_path,
-            out_inferred_pop_ht_path=inferred_pop_ht_path,
-            out_sample_qc_ht_path=ancestry_sample_qc_ht_path,
-        )
-        ancestry_plots.run(
-            out_path_pattern=res_pref / 'plots' / '{scope}_pc{pci}_{pca_suffix}.{ext}',
-            sample_qc_ht_path=ancestry_sample_qc_ht_path,
-            scores_ht_path=scores_ht_path,
-            eigenvalues_ht_path=eigenvalues_ht_path,
-            loadings_ht_path=loadings_ht_path,
-            inferred_pop_ht_path=inferred_pop_ht_path,
-            relateds_to_drop_ht_path=relateds_to_drop_ht_path,
-        )
+def test_sample_qc(mocker: MockFixture, tmp_path: Path):
+    conf = create_config(tmp_path)
+    set_config(
+        conf,
+        tmp_path / 'config.toml',
+        merge_with=[DEFAULT_CONFIG, LARGE_COHORT_CONFIG],
+    )
 
-        siteonly_vcf_path = res_pref / 'siteonly.vcf.bgz'
-        site_only_vcf.run(
-            vds_path=str(vds_path),
-            sample_qc_ht_path=str(sample_qc_ht_path),
-            relateds_to_drop_ht_path=str(relateds_to_drop_ht_path),
-            out_vcf_path=str(siteonly_vcf_path),
-            tmp_prefix=str(res_pref / 'tmp'),
-        )
+    mocker.patch(
+        'cpg_workflows.inputs.deprecated_create_cohort',
+        lambda: _mock_cohort(conf.workflow.dataset),
+    )
+
+    # skip can_reuse, implicit skip of existence checks
+    mocker.patch('cpg_workflows.large_cohort.combiner.can_reuse', lambda x: False)
```

Comment:
Do we actually need to create a cohort for all these downstream tests? It's very hard (for me) to work out what's going on with the whole `query_command` rubbish and what actually need to be passed to the `run` method.

Reply:
This is a fair point... I don't think so. Ideally all the interactions with cohorts/sequencing groups should be done in the Stages (so the code is run during the runtime when the `workflow` is initiated. The changes you made recently to `combiner` implement that logic. that makes the interface with the methods way easier (run QC on these samples, instead of *find some samples and run QC on them*), and makes the testing way simpler. 

From scraping off additional bits of code from this test suite, I've found another instance of `get_multicohort()` being called inside the child/job logic in SampleQC. I'd refactor this to take that info as input, instead of to be determined at runtime.

If you went a step further, the logic under test here usually takes no more than 2/3 distinct entries from configuration, often taking a default value. Those could easily be made into arguments, then the stages don't even need a live configuration.